### PR TITLE
label 4N

### PR DESCRIPTION
**Pull request by rpatel3001**

### CODE REVIEW

**Review by makrsmark — CHANGES_REQUESTED** [review]
Tests?

File: lib/plugins/Label_4N.test.ts [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+  expect(decodeResult.formatted.items[3].code).toBe('ALT');
+  expect(decodeResult.formatted.items[3].value).toBe('23000 feet');
+});
+
+test('decodes Label 4N variant 2B', () => {
+  const decoder = new MessageDecoder();
+  const decoderPlugin = new Label_4N(decoder);
+
+  // https://app.airframes.io/messages/3421601874
+  const text = '285,B,69005074-507,10/12,+36.081,-094.810,35014,002.3,ELP,SDF,SDF,17R/,17L/,0,0,,,,,,0,0,0,0,1,,,,,247.0,014.2,261.2,421A';
```
Comment (makrsmark) [inline lib/plugins/Label_4N.test.ts:43]: do we know what the 285 is in these? I'm not seeing it in the tests

File: lib/plugins/Label_4N.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        ResultFormatter.arrivalRunway(decodeResult, fields[11].split("/")[0]);
+        ResultFormatter.checksum(decodeResult, fields[32]);
+        let remaining = [];
+        const idxs = [0, 3, 4, 5, 6, 8, 9, 10, 11, 12, 32];
+        for (let i = 0; i < fields.length; i++) {
+            if (fields[i] !== "" && !idxs.includes(i)) {
+                remaining = remaining.concat(fields[i]);
+            }
+        } 
+        decodeResult.remaining.text = remaining.join(",");
```
Comment (makrsmark) [inline lib/plugins/Label_4N.ts:56]: a little indirect, but i won't stop you

File: lib/plugins/Label_4N.ts [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        decodeResult.remaining.text = [text.substring(2, 4), text.substring(19, 29)].join(" ");
+    } else if (fields.length === 33) {
+        // variant 2
+        decodeResult.raw.date = fields[3];
+        if (fields[1] === "B") {
+            ResultFormatter.position(decodeResult, {latitude: Number(fields[4]), longitude: Number(fields[5])});
+            ResultFormatter.altitude(decodeResult, fields[6]);
+        }
+        ResultFormatter.departureAirport(decodeResult, fields[8]);
+        ResultFormatter.arrivalAirport(decodeResult, fields[9]);
```
Comment (makrsmark) [inline lib/plugins/Label_4N.ts:46]: idx 10 is alternate airport and 12 is the alternate runway. I made a note of this in Label 10. maybe it's worth parsing.

**Review by makrsmark — APPROVED** [review]

File: lib/plugins/Label_4N.test.ts (hunk 1 above)
Comment (rpatel3001) [reply lib/plugins/Label_4N.test.ts:43]: not sure but I haven't seen it be anything else so I assumed it was a constant value and skipped it

File: lib/plugins/Label_4N.ts (hunk 2 above)
Comment (rpatel3001) [reply lib/plugins/Label_4N.ts:56]: i thought so too, open to suggestions

Comment (rpatel3001) [reply lib/plugins/Label_4N.ts]: pulled from label 10

File: lib/plugins/Label_4N.ts (hunk 3 above)
Comment (rpatel3001) [reply lib/plugins/Label_4N.ts:46]: added, you might want changes in the result_formatter bit

**Review by makrsmark — APPROVED** [review]

**Pull request merged**